go/proof.go

```diff
@@ -40,6 +40,25 @@ var TendermintSpec = &ProofSpec{
 	},
 }
 
+// SmtSpec constrains the format for SMT proofs (as implemented by github.com/celestiaorg/smt)
```

Comment:
It would be nice to have a test case (even json file, table test... something generated by a real SMT that we can test against).

Let's verify this works with your SMT proofs.

Reply:
Originally this was in the cosmos-sdk repo, but I moved it here to keep a single source of truth.

Test cases are [here](https://github.com/cosmos/cosmos-sdk/pull/10015/files#diff-751f1a02f13499f5243d1b120677047e55f0ed907934676b1de420fab81dbd77), in the code for the sdk PR. I don't have one with a failed verification yet, but I will add that. Should I add any tests within this repo?

Reply:
Please add some basic tests in this repo.
Happy to have it here, just tested here as well